main/main.cpp

```diff
@@ -1661,20 +1679,21 @@ bool Main::iteration() {
 	OS::get_singleton()->get_main_loop()->idle( step*time_scale );
 	message_queue->flush();
 
+	if(!disable_render_loop){
```

Comment:
This is probably pretty unsafe, if rendering was moved to another thread, the command ringbuffer will never be flushed. At least sync() needs to be called always.